### PR TITLE
Add post, comment and like forms

### PR DESCRIPTION
**Pull request by adamilare**
In this update the following were implemented;
- Add form for new post
- Add form for new comment
- Add form for new like
- 'current_user' can add post, comment, and like

### CODE REVIEW

File: app/views/shared/_post_item.html.erb [hunk 1]
```diff
@@ -32,4 +26,10 @@
             <% end %>
         </ul>
     <% end%>
+    <div class="add-comment" id="comment-form-<%= post.id %>">
+        <%= form_with model: @comment, method: :post, url: "/users/#{post.author.id}/posts/#{post.id}/comments", scope: 'comment' do |form| %>
+            <%= form.text_field :text, placeholder: "Add a comment...", class: "comment-text" %>
+            <%= form.submit "Add" %>
+        <% end %>
+  </div>
```
Comment (jerryowusu) [inline app/views/shared/_post_item.html.erb]: Nice one there. However, to make your project more professional and follow best standard practices, kindly create a new file, `views/comments/new.html.erb` and put the form that will accept inputs for creating comments. :pray:

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    @comment.post_id = params[:post_id]
+
+    if @comment.save
+      flash[:success] = 'Comment added successfully!'
+      redirect_to request.referrer
+    else
+      flash.now[:error] = 'comment creation failed!'
+    end
+  end
+end
```
Comment (jerryowusu) [inline app/controllers/comments_controller.rb]: [OPTIONAL]  In order to increase the security of data sent through the forms kindly consider using strong params to specify in the controller which parameters are accepted and used just as you have done in the posts controller. You can check [this link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md) for further explanation on how to use the params.

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @adamilare , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

noooo


## Highlights
Nice design :+1: 
Data set up sucessfully :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/shared/_post_item.html.erb (hunk 1 above)
Comment (adamilare) [reply app/views/shared/_post_item.html.erb]: @jerryowusu , thank you for the review.
Partial still needed to be used though, as new route is not used for comment and like.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @adamilare 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**